learning/tour-of-beam/frontend/lib/auth/method.dart

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:enum_map/enum_map.dart';
+
+part 'method.g.dart';
+
+@unmodifiableEnumMap
+enum AuthMethod {
```

Comment:
Could we instead just rely on which https://pub.dev/documentation/firebase_auth_platform_interface/latest/firebase_auth_platform_interface/AuthProvider-class.html is referenced with a particular button action?  If someone clicks the Google button, it initializes Auth flow via Google.  If someone clicks the Github button, it initializes Auth flow via Github.  The code only seems to use the AuthMethod.google.

Reply:
I've added the `firebase_auth_platform_interface` package and now `logIn` takes `AuthProvider` as an argument.